### PR TITLE
fealib: allow newlines in string literals (and skip over them)

### LINKED ISSUE #578: [feaLib] Lexer for double quote strings should not stop at newline?
I have fonts and fea files with a name table where some nameid strings have newlines in the string. According to the spec 

> A string is composed of 1-byte ASCII characters enclosed by ASCII double quote characters ("). Newlines embedded within the string are removed from the character sequence to be stored.

Looking at a ttx dump these strings have `&#13;` inserted for the newlines so ttx seems to have no issue, but loading the fea file with feaLib fails because in lexer.py there is `self.scan_until_('"\r\n')`
If I remove  `\r\n` it loads fine, but then test_string fails in lexer_test.py because it doesn't raise an Exception at the embedded newline. Maybe I'm missing something, but I don't see a reason for the `\r\n` in the double quote lexer.

### PR DESCRIPTION
**Pull request by anthrotype**
should fix #578 

/cc @brawer

### CODE REVIEW

File: Lib/fontTools/feaLib/lexer.py [hunk 1]
```diff
@@ -142,13 +142,24 @@ def next_(self):
...
+                                          location)
+                cur_char = self.text_[self.pos_]
                 self.pos_ += 1
-                return (Lexer.STRING, text[start + 1:self.pos_ - 1], location)
-            else:
-                raise FeatureLibError("Expected '\"' to terminate string",
-                                      location)
+                if cur_char == '"':
+                    strings.append(text[prev_pos:self.pos_ - 1])
+                    return (Lexer.STRING, "".join(strings), location)
```
Comment (brawer) [inline Lib/fontTools/feaLib/lexer.py]: Wouldn’t it be easier to do self.scan_util('"') and then replace \r and \n by '' ?

Comment (anthrotype) [inline Lib/fontTools/feaLib/lexer.py]: I can have a look tomorrow. If you want, you can go on and fix it yourself, no prob ;)

**Comment by anthrotype** [comment]
@brawer I simplified the code a bit, like you suggested, and added a test case. It should be good to go now. Feel free to add to it in case I missed something.

**Comment by brawer** [comment]
👍

**Pull request merged**